peering/models/models.py

```diff
@@ -589,6 +602,27 @@ def get_relationship_html(self):
             f'<span class="badge {badge_type}">{self.get_relationship_display()}</span>'
         )
 
+    def generate_service_ref(self) -> str:
```

Comment:
Avoid type annotation for now as we don't have them in the code (yet).

Use a full name function `reference` instead of `ref` for readability.